util/systemd/systemd_linux.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package systemd
+
+import (
+	"log"
+	"sync"
+
+	"github.com/mdlayher/sdnotify"
+)
+
+var getNotifyOnce struct {
+	sync.Once
+	v *sdnotify.Notifier
+}
+
+func notifier() *sdnotify.Notifier {
+	getNotifyOnce.Do(func() {
+		var err error
+		getNotifyOnce.v, err = sdnotify.New()
+		if err != nil {
+			log.Printf("systemd: systemd-notifier error: %v", err)
+		}
+	})
+	return getNotifyOnce.v
+}
+
+// Ready signals readiness to systemd. This will unblock service dependents from starting.
+func Ready() {
+	err := notifier().Notify(sdnotify.Ready, sdnotify.Statusf("service is listening for traffic"))
+	if err != nil {
+		log.Printf("systemd: error notifying systemd: %v", err)
```

Comment:
put this log and the line 58 one behind a sync.Once? I don't think we really need to see that the systemd socket isn't present more than once.